pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/assignment/segment/OfflineSegmentAssignment.java

```diff
@@ -99,49 +99,48 @@ public void init(HelixManager helixManager, TableConfig tableConfig) {
         _offlineTableName);
     LOGGER.info("Assigning segment: {} with instance partitions: {} for table: {}", segmentName, instancePartitions,
         _offlineTableName);
+    checkReplication(instancePartitions);
 
-    List<String> instancesAssigned;
-    if (instancePartitions.getNumReplicaGroups() == 1) {
+    List<String> instancesAssigned = assignSegment(segmentName, currentAssignment, instancePartitions);
+
+    LOGGER
```

Comment:
Will it print too many messages if there are many segments in a table?

Reply:
That is exactly the reason why the `assignSegment` is extracted into a helper method (line 128)